src/core/dbus-manager.c

```diff
@@ -2073,6 +2113,7 @@ const sd_bus_vtable bus_manager_vtable[] = {
         SD_BUS_METHOD("ResetFailed", NULL, NULL, method_reset_failed, SD_BUS_VTABLE_UNPRIVILEGED),
         SD_BUS_METHOD("ListUnits", NULL, "a(ssssssouso)", method_list_units, SD_BUS_VTABLE_UNPRIVILEGED),
         SD_BUS_METHOD("ListUnitsFiltered", "as", "a(ssssssouso)", method_list_units_filtered, SD_BUS_VTABLE_UNPRIVILEGED),
+        SD_BUS_METHOD("ListUnitsByStatesAndPatterns", "asas", "a(ssssssouso)", method_list_units_filtered_by_states_and_patterns, SD_BUS_VTABLE_UNPRIVILEGED),
```

Comment:
The name sounds a bit cumbersome. Please just call this "ListUnitsByPatterns" or so...